.github/workflows/release.yml

```diff
@@ -0,0 +1,96 @@
+name: Release
+on:
+  push:
+    tags: '*'
+
+# Needed to create the release
+permissions:
+  contents: write
+  id-token: write
+
+# Needed to create the release and use gh cli
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  GH_TOKEN:  ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  create-release:
+    runs-on: ubuntu-latest
+    steps:
+        - name: Checkout Code
+          uses: actions/checkout@v4 
+        - name: Create a Release
+          id: create_release
+          uses: actions/create-release@v1
+          with:
+            tag_name: ${{ github.ref }}
+            release_name: Release ${{ github.ref_name }}
+            prerelease: true
+
+  build-and-push-amd64:
+    runs-on: ubuntu-latest
+    needs: [create-release]
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Run Dapper CI
+        run: |
+          make 
+          ./.dapper ci
+      - name: Generate Checksum
+        run: sha256sum dist/artifacts/* > dist/artifacts/sha256sum-amd64.txt
+      - name: Upload Assets
+        run: |
+          ASSETS_DIR="dist/artifacts"
+          TAG=${{ github.ref_name }}
+          for file in $ASSETS_DIR/*; do
+            gh release upload $TAG $file --repo ${{ github.repository }}
+          done
+      - name: Build Manifest
+        id: metadata
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            docker.io/rancher/rancher-csp-adapter
+            709825985650.dkr.ecr.us-east-1.amazonaws.com/suse/rancher/rancher-csp-adapter
+            709825985650.dkr.ecr.us-east-1.amazonaws.com/suse/rancher/rancher-csp-adapter-eu
+          tags: |
```

Comment:
We need need to specify the `tags` section, just use the default one will do. See https://github.com/docker/metadata-action?tab=readme-ov-file#basic

Reply:
Are we pushing latest tag also? I have it disabled.

Reply:
No, we don't need the latest tag. Version tag should be fine.